File: eunbin/정수 제곱근 판별.cpp
#include <string>
#include <vector>
#include <cmath>
using namespace std;

//1. 제곱근 구함 2. 자연수 인지 확인 3. 맞으면 제곱 아니면 -1

    long long solution(long long n) {
    long long answer = 0;
    
    long long cand = sqrt(n);
    if(cand * cand == n) {
        answer = (cand+1)  * (cand+1);
    } else {
        answer = -1;
    }
    
    return answer;


}

File: bsw112/2021-01-27/정수제곱근판별.java
public class 정수제곱근판별 {

    public static void main(String[] args) {
        정수제곱근판별 module = new 정수제곱근판별();
        System.out.println(module.solution(121));
    }
    public long solution(long n) {
        double sqrt = Math.sqrt(n);
        if(sqrt % 1 != 0){
           return -1;
        }
        return (long)Math.pow(sqrt + 1, 2);
    }
}
